src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java

```
@@ -49,36 +335,45 @@ public void swap(int x1, int y1, int x2, int y2) {
         boolean[] res = new boolean[6];
+        Piece piece = getPiece(x, y);
         //check east
-        if (GRID_DIMENSIONS - x > 2) {
-            res[0] = getPiece(x, y).equals(getPiece(x + 1, y)) && getPiece(x, y).equals(getPiece(x + 2, y));
+        if (x < GRID_DIMENSIONS - 2) {
+            res[0] = piece.equals(getPiece(x + 1, y)) && piece.equals(getPiece(x + 2, y));
         }
 
         //check north
-        if (y >= 2) {
-            res[1] = getPiece(x, y).equals(getPiece(x, y - 1)) && getPiece(x, y).equals(getPiece(x, y - 2));
+        if (y > 1) {
+            res[1] = piece.equals(getPiece(x, y - 1)) && piece.equals(getPiece(x, y - 2));
         }
 
         //check west
-        if (x >= 2) {
-            res[2] = getPiece(x, y).equals(getPiece(x - 1, y)) && getPiece(x, y).equals(getPiece(x - 2, y));
+        if (x > 1) {
+            res[2] = piece.equals(getPiece(x - 1, y)) && piece.equals(getPiece(x - 2, y));
         }
 
         //check south
-        if (GRID_DIMENSIONS - y > 2) {
-            res[3] = getPiece(x, y).equals(getPiece(x, y + 1)) && getPiece(x, y).equals(getPiece(x, y + 2));
+        if (y < GRID_DIMENSIONS - 2) {
+            res[3] = piece.equals(getPiece(x, y + 1)) && piece.equals(getPiece(x, y + 2));
         }
 
         //check vertical
-        if (GRID_DIMENSIONS - y > 1 && y >= 1) {
-            res[4] = getPiece(x, y).equals(getPiece(x, y + 1)) && getPiece(x, y).equals(getPiece(x, y - 1));
+        if (y < GRID_DIMENSIONS - 1 && y > 0) {
+            res[4] = piece.equals(getPiece(x, y - 1)) && piece.equals(getPiece(x, y + 1));
         }
 
         //check horizontal
-        if (GRID_DIMENSIONS - x > 1 && x >= 1) {
-            res[5] = getPiece(x, y).equals(getPiece(x + 1, y)) && getPiece(x, y).equals(getPiece(x - 1, y));
+        if (x < GRID_DIMENSIONS - 1 && x > 0) {
+            res[5] = piece.equals(getPiece(x - 1, y)) && piece.equals(getPiece(x + 1, y));
         }
 
```

I think this should be refactored later to look nicer, but it at least needs an else which will break the program (throwing a RuntimeException would be fine) instead of failing silently when there is a mistake in one of the if conditions.